### DISCUSSION TITLE
Do we actually need forms?

### DISCUSSION THREAD

**Discussion by ankitagarwaladobe** [ONDC-Official/ONDC-FIS-Specifications#26] [General]
Doesn't form add additional hop and increases latency. BAP first needs to submit the form and then wait for the response and then make subsequent calls?
Any specific reason for having forms in API flow.

**Comment by 92shreyansh** [comment]
This is mainly to handle and communicate the dynamicness of the fields expected from the BAP from the BPP and to capture specifics of the flow which are over and beyond the base transactional APIs. 

We are considering differnt models for this for future but as of now forms are the way identified even in the beckn specification for such requirements.

**Reply by ankitagarwaladobe** [reply]
Is it possible if the BPPs send a form but the BAPs can actually return the response as part of the subsequent request itself. 
I am just trying to cut down on latency here.

How do we ensure idempotency with the form submission?

What happens if the borrower wants to resubmit the form? Would the same form can be submitted multiple times? Would the latest form replace the older form?

Security concern?
How does the BPP ensure that the forms submission is blocked after the workflow has progressed to a later stage. Trying to avoid an attack here.

**Reply by ankitagarwaladobe** [reply]
@92shreyansh 
IIUC the recommendation of forms usage in beckn [here](https://github.com/beckn/protocol-specifications/blob/draft/docs/BECKN-007-The-XInput-Schema.md) is not the same as this vary usecase.
Reason:

1. The beckn example assumes the BAP to show the same form with CSS modifcations to the end user and the fields are not standardised whereas in [this thread](https://github.com/ONDC-Official/ONDC-FIS-Specifications/discussions/5) we are standardising the fields.
2. In the last call, it was told that the BAP can collect/consolidate the required information from the user for multiple vendors and transmit as per the vendor which again doesn't go comply with beckn example above.

If we are planning to standardise the fields, then I would propose that the lenders return a list of field ids they want and the BAPs respond accordingly in the subsequent api calls.
